### PR TITLE
starspace: init at 8aee0a950aa607c023e5c91cff518bec335b5df5

### PR DESCRIPTION
**Pull request by mausch**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{lib, stdenv, fetchFromGitHub, boost165}:
+
+stdenv.mkDerivation rec {
+  pname = "starspace";
+  version = "8aee0a950aa607c023e5c91cff518bec335b5df5";
```
Comment (fabaff) [inline pkgs/applications/science/machine-learning/starspace/default.nix:5]: Some packagers do something like this to make the version more human-friendy.

```suggestion
    version = "20210117-${lib.strings.substring 0 7 rev}";
    rev = "8aee0a950aa607c023e5c91cff518bec335b5df5";
```

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "starspace";
+    rev = version;
```
Comment (fabaff) [inline pkgs/applications/science/machine-learning/starspace/default.nix:10]:
```suggestion
    inherit rev;
```

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 1 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in https://github.com/NixOS/nixpkgs/pull/109651/commits/f6b9aa723fcf68ccd270b5278ceeacd3e431afc3

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 2 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in https://github.com/NixOS/nixpkgs/pull/109651/commits/f6b9aa723fcf68ccd270b5278ceeacd3e431afc3

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{lib, stdenv, fetchFromGitHub, boost165}:
+
+stdenv.mkDerivation rec {
+  pname = "starspace";
+  version = "20210117-${lib.strings.substring 0 7 rev}";
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:5]: As per CONTRIBUTING.md:

```suggestion
  version = "unstable-2021-01-17";
```

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  rev = "8aee0a950aa607c023e5c91cff518bec335b5df5";
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:6]: Drop this

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "starspace";
+    inherit rev;
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:11]:
```suggestion
    rev = "8aee0a950aa607c023e5c91cff518bec335b5df5";
```

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{lib, stdenv, fetchFromGitHub, boost165}:
+
+stdenv.mkDerivation rec {
+  pname = "starspace";
+  version = "20210117-${lib.strings.substring 0 7 rev}";
+  rev = "8aee0a950aa607c023e5c91cff518bec335b5df5";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "starspace";
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:10]:
```suggestion
    repo = pname;
```

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{lib, stdenv, fetchFromGitHub, boost165}:
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, boost165 }:
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  rev = "8aee0a950aa607c023e5c91cff518bec335b5df5";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "starspace";
+    inherit rev;
+    sha256 = "0sc7a37z1skb9377a1qs8ggwrkz0nmpybx7sms38xj05b702kbvj";
+  };
+
+  nativeBuildInputs = [ boost165 ];
```
Comment (prusnak) [inline pkgs/applications/science/machine-learning/starspace/default.nix:15]: This seems like to belong rather to `buildInputs`.

Also are you use newer boost cannot be used? If yes, please provide a comment describing the issue when building with newer boost (i.e. `boost` package).

Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: > This seems like to belong rather to buildInputs.
Applied in f8b69fa6d2fcf54548795e3e04845f304efb3a30

> Also are you use newer boost cannot be used?
[The documentation for StarSpace explicitly mentions Boost 1.63](https://github.com/facebookresearch/StarSpace#requirements), which was deleted from nixpkgs some time ago. So I used 1.65 which seems to work. 
No idea if if works with a newer version but why would we want to deviate so much from the official requirements?

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 5 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in f8b69fa6d2fcf54548795e3e04845f304efb3a30

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 6 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in f8b69fa6d2fcf54548795e3e04845f304efb3a30

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 7 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in f8b69fa6d2fcf54548795e3e04845f304efb3a30

File: pkgs/applications/science/machine-learning/starspace/default.nix (hunk 4 above)
Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix]: Applied in f8b69fa6d2fcf54548795e3e04845f304efb3a30

**Review by prusnak — APPROVED** [review]

**Comment by mausch** [comment]
Thank you for the reviews @prusnak @fabaff :+1:

**Pull request merged**